update for 7.1.5

hgdubbe opened the pull request:
nothing final, but works fine for me

hgdubbe committed:
update for 7.1.5

nothing final, but works fine for me

hgdubbe committed:
Update protection.lua

heistcode commented:
If you want to track the revenge proc:
```
do
	local Revenge = false
	NeP.Listener:Add('RevengeProcStart', 'SPELL_ACTIVATION_OVERLAY_GLOW_SHOW', function(spellID)
		if spellID == 6572 then
			Revenge = true
		end
	end)

	NeP.Listener:Add('RevengeProcStop', 'SPELL_ACTIVATION_OVERLAY_GLOW_HIDE', function(spellID)
		if spellID == 6572 then
			Revenge = false
		end
	end)

	NeP.DSL:Register("revengeproc", function()
		return Revenge
	end)
end
```

hgdubbe commented:
nice, thanks :D

heistcode commented:
Besides that a couple of the other changes I made was to readd that Impending Victory line into the CR and add `{'/startattack', '!isattacking'},` to the top of the inCombat table below `{Keybinds},`. I also removed the Heroic Leap keybind (I just use a macro: /cast [@cursor] Heroic Leap). I changed lcontrol to be:
```
	{'Heroic Throw', 'keybind(lcontrol)&mouseover.range>8&mouseover.range<=30&mouseover.infront', 'mouseover'},
	{'Heroic Throw', 'keybind(lcontrol)&target.range>8&target.range<=30&target.infront'}
```

heistcode commented:
I'm leveling prot warrior (level 103 atm) so I'm just making adjustments as I go.

heistcode commented:
I might put in checks in the Heroic Throw lines to only use Heroic Throw on targets that are not targeting me since I usually just use that to pick up adds.

hgdubbe committed:
Update protection.lua

added revenge proc -> dps boost :)

heistcode commented:
```
	{'Heroic Throw', 'keybind(lcontrol)&mouseover.range>8&mouseover.range<=30&mouseover.infront&!mouseovertarget.isself', 'mouseover'},
	{'Heroic Throw', 'keybind(lcontrol)&target.range>8&target.range<=30&target.infront&!targettarget.isself'}
```

hgdubbe committed:
Update protection.lua

hgdubbe commented:
added that aautoattack-line, makes a lot of sense :)

hgdubbe committed:
working rotation, keybinds for leap and throw

working rotation, keybinds for leap and throw

keep in mind you need updated conditions.lua for this to work

hgdubbe committed:
also use revenge on high priority if freeproc

also use revenge on high priority if freeproc

hgdubbe commented:
redid the whole rotation based on actual simc prot warrior t19M (beta) profile. actually it doesnt work, somehow it stops the whole game for like 1/2 sec after beginning the fight and doesnt do anything but battle cry....maybe you could take a look into it if you find anything? i would really appreciate

http://en.file-upload.net/download-12243048/protection.lua.html

heistcode commented:
I'll take a look when I get home. You can add stuff like that to a gist here on github.

heistcode commented:
Just looking through it quick I'm surprised it doesn't look like tclap is in the single target rotation. I heard it would be.

hgdubbe commented:
yep me too, once this thing runs we'll figure out :)

heistcode commented:
Nah. Don't bother with that. If they release a stable sim profile maybe consider a rewrite but I'm not going to troubleshoot something if I can tell it's already wrong. Tclap is supposed to be used on single target according to http://www.mmo-champion.com/threads/2030070-7-0-Guide-Protection-Back-with-Vengeance?p=41445768#rotation

hgdubbe commented:
ok, thanks anyway
old d4ef437 works well enough until then ;)

hgdubbe commented:
http://www.mmo-champion.com/threads/1892699-Protection-legion-Discussion?p=44124013&viewfull=1#post44124013
following this guide there is plenty of room for optimisation when delaying tc and revenge (the rotation works *very* well by now imho)...do you know any easy way to compare the gcd against the swing timer?
